sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java

```diff
@@ -40,7 +40,7 @@ static BlobOutputStream appendBlobOutputStream(final AppendBlobAsyncClient clien
         return new AppendBlobOutputStream(client, appendBlobRequestConditions);
     }
 
-    static BlobOutputStream blockBlobOutputStream(final BlockBlobAsyncClient client,
+    public static BlobOutputStream blockBlobOutputStream(final BlobAsyncClient client,
```

Comment:
I had to make this public since the crypto package needs access. I dont see a problem doing this.

Reply:
I don't have a strong issue with it, we may want to split this and the other implementations into their own class files and make them public as well. I don't see a overwhelmingly strong reason why a customer couldn't pass a client in themselves to create the stream vs needing to get it from the client.

Reply:
I can do this as part of another issue I'm working on with Append and Page Blob Output Stream